rest-service/manager_rest/security/authorization.py

```diff
@@ -17,31 +16,27 @@
 
 
 def authorize(action,
-              tenant_for_auth=None,
               get_tenant_from='header',
               allow_all_tenants=False,
               allow_if_execution=False):
     def authorize_dec(func):
         @wraps(func)
         def wrapper(*args, **kwargs):
-            execution = execution_token.current_execution
-            if execution and allow_if_execution:
-                utils.set_current_tenant(execution.tenant)
-                return func(*args, **kwargs)
-
-            # getting the tenant name
+            tenant_name = None
             if get_tenant_from == 'header':
-                tenant_name = tenant_for_auth or request.headers.get(
-                    CLOUDIFY_TENANT_HEADER)
+                tenant_name = request.headers.get(CLOUDIFY_TENANT_HEADER)
             elif get_tenant_from == 'param':
-                tenant_name = tenant_for_auth or kwargs['tenant_name']
+                tenant_name = kwargs['tenant_name']
             elif get_tenant_from == 'data':
-                tenant_name = tenant_for_auth or get_json_and_verify_params(
+                tenant_name = get_json_and_verify_params(
                     {'tenant_name': {'type': text_type}}).get('tenant_name')
-            else:
-                tenant_name = tenant_for_auth
 
-            # finding tenant to add to the app config
+            execution = execution_token.current_execution
+            if execution and allow_if_execution:
+                if execution.tenant.name == tenant_name or not tenant_name:
+                    utils.set_current_tenant(execution.tenant)
+                    return func(*args, **kwargs)
+
```

Comment:
```suggestion
            if allow_if_execution:
                if current_execution and (
                    current_execution.tenant.name == tenant_name
                    or not tenant_name
                ):
                    utils.set_current_tenant(current_execution.tenant)
                    return func(*args, **kwargs)
```
(plus change the import to `from manager_rest.execution_token import current_execution`)
I suggest this way:
1. The whole special-cased code is now only under the `allow_if_execution`, which is the rare case
2. Tbh naming `execution = current_execution` is ugly: `current_execution` is a proxy and keeping that name helps reminds us of that

Reply:
OK, reworking this into a separate commit to include the updated import.